pkg/innerring/state.go

```diff
@@ -128,7 +128,11 @@ func (s *Server) voteForFSChainValidator(validators keys.PublicKeys, trigger *ut
 	if trigger != nil {
 		nonce, vub, err = s.morphClient.CalculateNonceAndVUB(*trigger)
 		if err != nil {
-			return fmt.Errorf("could not calculate nonce and `validUntilBlock` values: %w", err)
+			s.log.Debug("could not calculate nonce and `validUntilBlock` values", zap.Error(err))
+			nonce, vub, err = s.morphClient.CalculateNonceAndVUB(util.Uint256{})
```

Comment:
commit states
> ... an error is returned because there is no hash in the FSchain. Ignore the error and calculate the value in another way

but we switch to zero hash on any failure. Shouldnt we check the exact case?

Reply:
IIUC we should _always_ call `CalculateNonceAndVUB` with `util.Uint256{}` in `voteForFSChainValidator` if `trigger == nil`, because the only user of this method (not counting validator predefined in config) is governance processor reacting to main chain events.